src/utils/commonUtils.js

```diff
@@ -171,4 +171,16 @@ module.exports = {
     return crypted;
   }, // encrypt
 
+  /**
+   * Determine if the given Generator is set up for bulk collection
+   * @param  {Object} generator - A Generator object
+   * @returns {Boolean}
+   */
+  isBulk(generator) {
+    const gt = generator.generatorTemplate;
+    const connection = gt && gt.connection;
+    const bulk = connection && connection.bulk;
+    return Boolean(bulk);
```

Comment:
if the "generator" arg doesn't contain a generatorTemplate, it's an error... just returning bulk=false seems wrong here

if the generatorTemplate doesn't contain a connection, it's an error... just returning bulk=false seems wrong here

if the connection doesn't contain bulk, then yes, we treat it as the default bulk=false

Reply:
I did it this way because it makes it easier to mock generator templates for testing and there were already tests that relied on it not throwing an error.
I could change the tests but I don't think this is the right place to enforce the format. A real generator template should already be validated by this point, right?